earthpy/tests/test_mask.py

```diff
@@ -1,52 +1,92 @@
 """ Tests for the mask module. """
 
 import numpy as np
-from earthpy.mask import make_apply_mask
+from earthpy.mask import mask_pixels
 import pytest
 
 
-im = np.random.randint(10, size=(4, 5))
-im_mask = np.random.randint(5, size=(4, 5))
+@pytest.fixture
+def im():
+    """Create image arr for masking"""
+    return np.random.randint(10, size=(4, 5))
 
 
-def test_arr_provided():
-    """ Test for arrays provided as image and/or mask input. """
+@pytest.fixture
+def im_mask():
+    """Create image mask"""
+    return np.random.randint(5, size=(4, 5))
+
+
+def test_arr_provided(im, im_mask):
+    """ Test that inputs are numpy arrays. """
 
     an_arr_tup = (2, 3)
-    with pytest.raises(AttributeError):
-        make_apply_mask(an_arr_tup, mask_arr=im_mask, vals=[0, 4])
-    with pytest.raises(AttributeError):
-        make_apply_mask(im, mask_arr=an_arr_tup, vals=[0, 4])
+    with pytest.raises(
+        AttributeError, match="Input arr should be a numpy array."
+    ):
+        mask_pixels(an_arr_tup, mask_arr=im_mask, vals=[0, 4])
+    with pytest.raises(
+        AttributeError, match="Input arr should be a numpy array."
+    ):
+        mask_pixels(im, mask_arr=an_arr_tup, vals=[0, 4])
 
 
-def test_masked_arr_returned():
+def test_masked_arr_returned(im, im_mask):
     """ Test for return of masked_array type. """
 
-    masked = make_apply_mask(im, im_mask, [0, 4])
+    masked = mask_pixels(im, im_mask, vals=[0, 4])
     assert np.ma.is_masked(masked)
 
 
-def test_val_list_provided():
-    """ Provide a singular value to the vals keyword to raise AttributeError"""
+def test_vals_as_list(im, im_mask):
+    """ Test for return of masked_array type. """
+
+    with pytest.raises(
+        AttributeError, match="Values should be provided as a list"
+    ):
+        mask_pixels(im, mask_arr=im_mask, vals=(4))
 
-    with pytest.raises(AttributeError):
-        make_apply_mask(im, mask_arr=im_mask, vals=0)
 
+def test_mask_contains_ones(im, im_mask):
+    """A boolean mask without the value 1 fails gracefully."""
 
-def test_mask_provided_with_no_ones():
-    """ Test that a mask is provided to apply mask that doesn't have 1's in it.
-        should return - there's nothing to mask here!"""
+    im_mask[im_mask > 0] = 0
+    with pytest.raises(ValueError, match="Mask requires values of 1"):
+        mask_pixels(im, mask_arr=im_mask)
+
+
+def test_mask_vals_in_arr(im, im_mask):
+    """If vals are not found in mask_arr, fail gracefully"""
 
     im_mask_no_ones = im_mask * 0
-    with pytest.raises(ValueError):
-        make_apply_mask(im, im_mask_no_ones, [3, 4])
+    with pytest.raises(
+        ValueError,
+        match="""List of values provided for the mask does 
+                             not exist in your mask array.""",
+    ):
+        mask_pixels(im, im_mask_no_ones, [3, 4])
+
+
+def test_masked_arr_provided(im, im_mask):
```

Comment:
This test is pretty hard to reason about because both `im` and `im_mask` are random. The two random numbers in particular (2 and 4) make it difficult to understand. For example, the line

```python
masked_im = np.ma.masked_where(im_mask < 2, im)
```

is hard to grok, because there is a chance that there are no places where `im_mask < 2`.

As far as I can see, there's no downside to having the `im` and `im_mask` fixtures be deterministic instead of random.

Reply:
ok @mbjoseph so you are suggesting that i hard code those two arrays rather than create one like this: `np.random.randint(10, size=(4, 5))` no problem.